modules/oe_migration_workflow/src/Plugin/migrate/process/SetWorkflowState.php

```diff
@@ -0,0 +1,259 @@
+    }
+
+    return $value;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   *   Thrown if the entity type doesn't exist.
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   *   Thrown if the storage handler couldn't be loaded.
+   */
+  public function validateConfigurationKeys(array $keys = NULL): void {
+    // Check if the configuration options are set and are strings.
+    foreach (['workflow_config_name', 'published_state', 'unpublished_state'] as $option) {
+      if (!array_key_exists($option, $this->configuration) || !is_string($this->configuration[$option])) {
+        /** @var string $message */
+        $message = sprintf('The "%s" option must be a string. The given value is of type "%s".', $option, gettype(($this->configuration[$option])));
+        throw new \InvalidArgumentException($message);
+      }
+    }
+    // The getWorkflow() method might throw an exception if the entity type
+    // doesn't exist or if the storage handler couldn't be loaded.
+    $workflow = $this->getWorkflow($this->configuration['workflow_config_name']);
+    $workflow_config_name = explode('.', $this->configuration['workflow_config_name']);
+    // Check if the given workflow is a valid one.
+    if (is_null($workflow)) {
+      $message = sprintf('"%s" is not a valid workflow.', end($workflow_config_name));
+      throw new \InvalidArgumentException($message);
+    }
+    // Check if the published state is a valid state for the given workflow.
+    elseif ($this->isValidWorkflowState($workflow, $this->configuration['published_state']) === FALSE) {
+      $message = sprintf('"%s" is not a valid state of the "%s" workflow.', $this->configuration['published_state'], end($workflow_config_name));
+      throw new \InvalidArgumentException($message);
+    }
+    // Check if the unpublished state is a valid state for the given workflow.
+    elseif ($this->isValidWorkflowState($workflow, $this->configuration['unpublished_state']) === FALSE) {
+      $message = sprintf('"%s" is not a valid state of the "%s" workflow.', $this->configuration['unpublished_state'], end($workflow_config_name));
+      throw new \InvalidArgumentException($message);
+    }
+  }
+
+  /**
+   * Get the workflow entity object from the configuration name.
+   *
+   * @param string $config_name
+   *   The configuration object name.
+   *
+   * @return \Drupal\workflows\WorkflowInterface|null
+   *   A workflow entity object. NULL if no matching entity is found.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   *   Thrown if the entity type doesn't exist.
+   * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+   *   Thrown if the storage handler couldn't be loaded.
+   */
+  protected function getWorkflow(string $config_name): ?WorkflowInterface {
```

Comment:
Wow, it's true. It simplifies a lot the code and related test. I performed a huge refinement of the code refactoring this part and allowing only the config as a workflow config (eg. `oe_corporate_workflow`). Less dependencies and same functionality.

Reply:
This is totally true, but I think that the `getStorage` call should be placed in the static `create` method and probably the `$workflow` variable of the `transfom()` method should become a class variable, since it's going to be shared by every row. As it is now, it's called for every row. Mea culpa, sorry.

Reply:
I changed the code for adding the getStorage call in the create method (you're right, it's better) but I'm not sure about the other point (to set the $workflow var as a class property) so I think that for the moment we can continue without change it. BTW obviously we can change it in the future.

Reply:
I have to add that this little change (move the `getStorage` method to the create method) improve and decrease the test complexity removing a (now useless) mock :D